web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/controller/ValidationController.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 Jarvus Innovations LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mobilitydata.gtfsvalidator.web.service.controller;
+
+import com.google.cloud.storage.*;
+import java.io.*;
+import java.net.URL;
+import java.util.*;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import org.mobilitydata.gtfsvalidator.web.service.util.StorageHelper;
+import org.mobilitydata.gtfsvalidator.web.service.util.ValidationHandler;
+import org.mobilitydata.gtfsvalidator.web.service.util.ValidationJobMetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+public class ValidationController {
+
+  private final Logger logger = LoggerFactory.getLogger(ValidationController.class);
+
+  @Getter(AccessLevel.PROTECTED)
+  @Setter(AccessLevel.PROTECTED)
+  @Autowired
+  private Storage storage;
+
+  @Autowired private ApplicationContext applicationContext;
+
+  /**
+   * Creates a new job id and returns it to the client. If a url is provided, the file is downloaded
+   * from the url and saved to GCS. If no url is provided, a unique url is generated for the client
+   * to upload the GTFS file.
+   */
+  @CrossOrigin(origins = "*")
+  @PostMapping(value = "/create-job", produces = "application/json", consumes = "application/json")
+  public CreateJobResponse createJob(@RequestBody CreateJobRequest body) {
+    try {
+      final var jobId = UUID.randomUUID().toString();
+      var countryCode = "";
+      StorageHelper storageHelper = new StorageHelper(storage, applicationContext);
```

Comment:
Could this be @AutoWired as well?

Reply:
@bdferris-v2 I'm not very experienced with SpringBoot and/or Autowiring (not even 100% sure they are related!).  I tried just making it an `@Autowired` property and removing the instantiation, but got errors.  Would need guidance if this is considered an important change [see comment re: prioritization of remaining changes](https://github.com/MobilityData/gtfs-validator/pull/1317#pullrequestreview-1357581006)

Reply:
Adding a @Component (aka org.springframework.stereotype.Component) to the `StorageHelper` class will make it auto-wireable by Spring.  I've verified that the application will run without complaints if you switch to @Autowired injection of StorageHelper ala the following patch:

```
diff --git a/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/controller/ValidationController.java b/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/controller/ValidationController.java
index a25aa498..6253c7ac 100644
--- a/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/controller/ValidationController.java
+++ b/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/controller/ValidationController.java
@@ -29,7 +29,6 @@ import org.mobilitydata.gtfsvalidator.web.service.util.ValidationJobMetaData;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.ApplicationContext;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
@@ -40,12 +39,8 @@ public class ValidationController {
 
   private final Logger logger = LoggerFactory.getLogger(ValidationController.class);
 
-  @Getter(AccessLevel.PROTECTED)
-  @Setter(AccessLevel.PROTECTED)
   @Autowired
-  private Storage storage;
-
-  @Autowired private ApplicationContext applicationContext;
+  private StorageHelper storageHelper;
 
   /**
    * Creates a new job id and returns it to the client. If a url is provided, the file is downloaded
@@ -57,8 +52,6 @@ public class ValidationController {
   public CreateJobResponse createJob(@RequestBody CreateJobRequest body) {
     try {
       final var jobId = UUID.randomUUID().toString();
-      StorageHelper storageHelper = new StorageHelper(storage, applicationContext);
-
       if (body != null && !Strings.isNullOrEmpty(body.getCountryCode())) {
         storageHelper.saveJobMetaData(jobId, body.getCountryCode());
       }
@@ -98,7 +91,6 @@ public class ValidationController {
       var jobId = jobData.getJobId();
       var fileName = jobData.getFileName();
 
-      StorageHelper storageHelper = new StorageHelper(storage, applicationContext);
       var countryCode = storageHelper.getJobCountryCode(jobId);
 
       // copy the file from GCS to a temp directory
diff --git a/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/util/StorageHelper.java b/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/util/StorageHelper.java
index 7613b1be..fe7330ac 100644
--- a/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/util/StorageHelper.java
+++ b/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/util/StorageHelper.java
@@ -12,8 +12,10 @@ import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
 
 /** Helper class for interacting with GCS. */
+@Component
 public class StorageHelper {
   static final String JOB_INFO_BUCKET_NAME = "gtfs-validator-results";
   static final String JOB_FILENAME_PREFIX = "job";
```

Reply:
Thank you!